luarules/gadgets/unit_attached_con_turret_mex.lua

```diff
@@ -0,0 +1,115 @@
+function gadget:GetInfo()
+    return {
+        name      = 'Legion Con Turret Metal Extractor',
+        desc      = 'Allows the mex to function as a con turret by replacing it with a fake mex with a con turret attached',
+        author    = 'EnderRobo',
+        version   = 'v1',
+        date      = 'September 2024',
+        license   = 'GNU GPL, v2 or later',
+        layer     = 12,
+        enabled   = true
+    }
+end
+
+if not gadgetHandler:IsSyncedCode() then
+    return false
+end
+
+local SpGetUnitPosition = Spring.GetUnitPosition
+local SpGetUnitDefID = Spring.GetUnitDefID
+local SpGetUnitAllyTeam = Spring.GetUnitAllyTeam
+local SpGetUnitHealth = Spring.GetUnitHealth
+local SpGetUnitBuildFacing = Spring.GetUnitBuildFacing
+local SpGetUnitIsDead = Spring.GetUnitIsDead
+local SpGetUnitStates = Spring.GetUnitStates
+
+function gadget:UnitFinished(unitID, unitDefID, unitTeam)
+
+	local unitDef = UnitDefs[unitDefID]
+	if unitDef.name == "legmohocon" then																-- checks for mex
```

Comment:
Seems like a weird way to do this in my opinion, wouldn't storing the def id
```lua
local legmohoconDefID = UnitDefNames[legmohocon] and UnitDefNames[legmohocon].id
```

and then comparing to that

```lua
if unitDefID ~= legmohoconDefID  then return end
```

be better? otherwise you have to preform a random read on every unit finished, and then get it's name which doesn't sound cache efficent to me.

Reply:
Done.